src/Native/Runtime/unix/PalRedhawkUnix.cpp

```diff
@@ -201,53 +212,55 @@ class UnixEvent
       m_state(initialState)
     {
         int st = pthread_mutex_init(&m_mutex, NULL);
-        ASSERT(st == NULL);
+        FATAL_ASSERT(st == 0, "Failed to initialize UnixEvent mutex");
```

Comment:
Right, all events seem to be created during runtime startup today, and so handling the failures using fail fast is not too bad. If we have propagated the failure up, the net result is going to be about the same today.

My point was more from the PAL contract point of view. The create event PAL API can return errors, so why not just return error.